src/webview/MessageList.js

```diff
@@ -105,34 +105,32 @@ export type Props = {|
 |};
 
 /**
- * The path to the webview assets, represented as a `file:`-scheme URL.
+ * The path to the webview assets, represented as a `file:`-schema URL. This may
+ * be a relative URL; if so, it is relative to the bundle or package root, and
```

Comment:
> the bundle or package root

I just spent a few minutes studying to see if I can unpack what this means. I think what's meant here is
* on iOS, the root directory of the app's bundle
  * `foo.bundleURL` is ["[t]he full URL of the receiver’s bundle directory"](https://developer.apple.com/documentation/foundation/bundle/1415654-bundleurl)
  * Looking in particular into what "bundle" means -- it's apparently [a pretty generic Apple concept](https://developer.apple.com/documentation/foundation/bundle). There might be several bundles floating around; `Bundle.main` gets specifically the one that contains our app.
* on Android... not sure. What's "the package root" mean concretely? And if we did use a relative URL here, would it get interpreted relative to that?
  * If it isn't trivial to definitively answer this, probably cleanest to just not try to generalize the description in that direction.

Reply:
> An iOS app bundle and an Android APK are, on some level, the same thing – they're a directory structure with a bunch of files: some executable, some resources, some metadata. An APK is a zip file of that structure, while... well, I assume there's some similar packaging mechanism for iOS, but I have no idea how it's customarily serialized.
> 
> What I mean here is the root of that directory structure, _as it is mounted at runtime_.

Yeah, so in the Android case I think there's a difference between
* the APK, which is a zip file used as a distribution format for the app
* the various directories belonging to the app once installed

I've been fuzzy on the details... so I went and investigated, details [in chat](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/android.20filesystem/near/796411).

One thing that seems to come out of that is that `/android_asset/` doesn't seem to be any kind of directory in the real filesystem at all! My best guess is that URLs under `file:///android_asset/` are specially interpreted by the WebView implementation (meaning the underlying Android WebView.)

And in comparing the iOS and Android cases, it looks like:
* an iOS bundle is an actual directory
* but on Android when an app is installed, the APK mostly doesn't get unpacked, nor mounted as a filesystem -- so with a few exceptions, there's no actual directory in the filesystem where the files inside it live. Instead, you get at the things inside it through a variety of APIs like [`Resources`](https://developer.android.com/reference/android/content/res/Resources.html) and [`AssetManager`](https://developer.android.com/reference/android/content/res/AssetManager).

Reply:
I'm somewhat surprised that APKs are never mounted into the underlying filesystem proper... but ultimately I see that as an irrelevant implementation detail. The filesystem seen by the WebView (or WebView-interpreted code) under `file://` may be mostly or entirely virtual, but it's still a (read-only) filesystem.

> My best guess is that URLs under `file:///android_asset/` are specially interpreted by the WebView implementation (meaning the underlying Android WebView.)

I can't find real documentation for that, but there are a couple of Chromium bugs where developers have given offhanded confirmation of this:
* https://bugs.chromium.org/p/chromium/issues/detail?id=346718
* https://bugs.chromium.org/p/chromium/issues/detail?id=387898

Reply:
> I can't find real documentation for that, but there are a couple of Chromium bugs where developers have given offhanded confirmation of this:

Cool -- I copied that finding over to the linked chat thread :)